android/modules/ui/src/java/ti/modules/titanium/ui/widget/webview/TiWebViewClient.java

```diff
@@ -187,4 +191,39 @@ public void onReceivedSslError(WebView view, SslErrorHandler handler, SslError e
 			Log.e(TAG, "SSL error occurred: " + error.toString());
 		}
 	}
+
+	@Override
+	public void onLoadResource(WebView view, String url)
+	{
+		super.onLoadResource(view, url);
+		KrollDict data = new KrollDict();
+		data.put(TiC.PROPERTY_URL, url);
+		webView.getProxy().fireSyncEvent(TiC.EVENT_WEBVIEW_ON_LOAD_RESOURCE, data);
+	}
+
+	// This method was deprecated from Android API level 21.
+	@TargetApi(Build.VERSION_CODES.HONEYCOMB)
+	@Override
+	public WebResourceResponse shouldInterceptRequest(WebView view, String url)
+	{
+		if(Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
```

Comment:
'shouldInterceptRequest' event seems to return the same thing as 'onloadResource' event. Why do we need to fire 2 events back to back if they give the same information? Perhaps I'm missing something here? Also 'shouldInterceptRequest' event is deprecated in Lollipop, so I'm not sure we should add a feature that is already deprecated.